experiments/ClimaEarth/components/land/climaland_integrated.jl

```diff
@@ -166,18 +167,33 @@ function ClimaLandSimulation(
     orog_adjusted_T = CC.Fields.Field(orog_adjusted_T_data, subsurface_space)
     orog_adjusted_T_surface = CC.Fields.Field(CC.Fields.level(orog_adjusted_T_data, 1), surface_space)
 
-    # Set initial conditions for the state
-    @. Y.soil.ϑ_l = θ_r + (ν - θ_r) / 2
-    Y.soil.θ_i .= FT(0.0)
-    ρc_s = CL.Soil.volumetric_heat_capacity.(Y.soil.ϑ_l, Y.soil.θ_i, soil_args.parameters.ρc_ds, earth_param_set)
-    Y.soil.ρe_int .= CL.Soil.volumetric_internal_energy.(Y.soil.θ_i, ρc_s, orog_adjusted_T, earth_param_set)
+    # Set initial conditions that aren't read in from file
     Y.soilco2.C .= FT(0.000412) # set to atmospheric co2, mol co2 per mol air
     Y.canopy.hydraulics.ϑ_l.:1 .= canopy_component_args.hydraulics.parameters.ν
     @. Y.canopy.energy.T = orog_adjusted_T_surface
 
-    Y.snow.S .= FT(0)
-    Y.snow.S_l .= FT(0)
-    Y.snow.U .= FT(0)
+    # Read in initial conditions for snow and soil from file, if requested
+    if land_spun_up_ic
+        # Set snow T first to use in computing snow internal energy
+        @. p.snow.T = orog_adjusted_T_surface
+
+        # Read in initial conditions for snow and soil
+        ic_path = CL.Artifacts.soil_ic_2008_50m_path()
+        CL.Simulations.set_snow_initial_conditions!(Y, p, surface_space, ic_path, model.snow.parameters)
+
+        T_bounds = extrema(Y.canopy.energy.T)
+        CL.Simulations.set_soil_initial_conditions!(Y, ν, θ_r, subsurface_space, ic_path, model.soil, T_bounds)
+    else
+        # Set initial conditions for the state
+        @. Y.soil.ϑ_l = θ_r + (ν - θ_r) / 2
```

Comment:
All of Y.soil is set in the function set_soil_initial_conditions!, so we dont need these lines